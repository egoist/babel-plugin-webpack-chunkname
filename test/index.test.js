const babel = require('babel-core')

const plugin = require.resolve('../')

test('main', () => {
  const { code } = babel.transform(`import('foo')\nimport('./foo.js')\nimport(\`./foo/\${bar}\`)`, {
    plugins: [
      'syntax-dynamic-import',
      plugin
    ]
  })
  expect(code).toMatchSnapshot()
})
