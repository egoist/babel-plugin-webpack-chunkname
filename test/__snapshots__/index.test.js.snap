// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom 1`] = `
"import(
/* webpackChunkName: 'foo-bar' */
'./foo/bar');"
`;

exports[`main 1`] = `
"import(
/* webpackChunkName: 'foo' */
'foo');
import(
/* webpackChunkName: 'foo' */
'./foo.js');
import(
/* webpackChunkName: 'foo/[request]' */
\`./foo/\${bar}\`);"
`;

exports[`skip exising magic comment 1`] = `
"import(
/* webpackChunkName: 'existing' */
'./foo');"
`;
